### DISCUSSION TITLE
Datenübertragung nach dem Booten

### DISCUSSION THREAD

**Discussion by BerndR329** [jomjol/AI-on-the-edge-device#2406] [Q&A]
Ich habe Probleme bei der Datenübertragung mit dem ESP32-Cam Modul. Ich habe das Modul mit dem Web-Installer geflasht und die SD-Karte entsprechend vorbereitet. Das Log beim Booten erscheint normal, ohne Fehler bis auf die Datenübertragung am Schluss des Bootens. Das führt dazu, dass ich nur teilweise das Setup-Menü sehe.
Am Ende des Boot-Logs sieht es so aus:

```
I (1013625) OTA FILE: uri: /log
I (1013635) OTA FILE: File contains 758 bytes
W (1022775) httpd_txrx: httpd_sock_err: error in recv : 104
W (1033805) httpd_txrx: httpd_sock_err: error in send : 11
E (1033805) SERVER HELP: File sending failed!
W (1038805) httpd_txrx: httpd_sock_err: error in send : 11
W (1038805) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to send file
W (1043805) httpd_txrx: httpd_sock_err: error in send : 11
W (1048805) httpd_txrx: httpd_sock_err: error in send : 11
W (1048805) httpd_uri: httpd_uri: uri handler execution failed
```

Hat jemand eine Idee, wo ich den Fehler suchen könnte?

viele Grüße
Bernd

**Comment by caco3** [comment]
Passiert das nach einem Restart und Web UI Reload immer noch?

Die gezeigten Fehler sind nicht kritisch.

**Comment by BerndR329** [comment]
Ja, es passiert immer noch in unterschiedlicher Ausprägung. Ein Kanalwechsel auf der Fritzbox 7390 von "auto" auf festen Kanal hat eine erhebliche Verbesserung gebracht. Im Moment teste ich die Auswirkung einer externen Antenne. Scheint damit auch besser zu werden.
Aber ich sehe die Fehler im Log bezgl Datenübertragung immer noch.
Das ist nicht kritisch? Machen das viele ESP32-Cam-Boards? Das habe ich mit anderen WLAN Geräten nicht erlebt, allerdings beobachtet man auch nicht die Logs wenn alles geht.

**Reply by caco3** [reply]
Errors in der HTTP-Kommunikation sind doof, beeinträchtigen die Funktionalität des Gerätes jedoch nicht.

Die Daten werden trotzdem (falls aktiviert) auf die SD-Karte gespiechert. Und die Datenübertragungen (MQTT/InfluxDB) sind unabhängig von den errors des Webservers.

**Comment by kaiduerkop** [comment]
Ich habe ebenfalls den o.g. Fehler und ich schaffe es nicht die GUI im Browser zu öffnen.
Ich habe es jetzt 3 verschiedenen ESP Cam probiert und mit 2 verschiedenen SD Karten.

```
I (12817) WIFI: Connected to: xxx, RSSI: -44
I (12827) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (13427) wifi:<ba-add>idx:0 (ifx:0, 26:5a:4c:15:46:95), tid:0, ssn:1, winSize:64
I (13817) esp_netif_handlers: sta ip: x.x.x.x, mask: 255.255.255.0, gw: x.x.x.x
I (13817) WIFI: Assigned IP: x.x.x.x
I (16597) SNTP: =================================================
I (16617) SNTP: ==================== Start ======================
I (16627) SNTP: == Logs before time sync -> log_1970-01-01.txt ==
I (16637) SNTP: Time is synced with NTP Server pool.ntp.org: 2024-01-29 01:37:44
I (17057) MAIN: Device info: CPU cores: 2, Chip revision: 301
I (17067) MAIN: SD card info: Name: ASTC, Capacity: 15360MB, Free: 987MB
I (19077) MAIN SERVER: Starting server on port: '80'
I (19077) MAIN SERVER: Registering URI handlers
I (19077) MAINCTRL: server_main_flow_task - Registering URI handlers
I (19077) OTA: Registering URI handlers
I (19077) GPIO: start GpioHandler
I (19087) GPIO: register GPIO Uri
I (19087) GPIO: server_GPIO - Registering URI handlers
I (19097) MAIN: Initialization completed successfully
I (19127) ov2640: Set PLL: clk_2x: 0, clk_div: 0, pclk_auto: 0, pclk_div: 8
I (21347) LOGFILE: Set log level to ERROR
I (21357) GPIO: read GPIO config and init GPIO
I (21377) GPIO: GPIO init completed, handler is disabled
I (21377) MAINCTRL: We are in Setup Mode -> Not starting Auto Flow!
I (30887) MAINCTRL: Autostart is not enabled -> Not starting Flow
W (173047) httpd_txrx: httpd_sock_err: error in send : 11
E (173047) SERVER HELP: File sending failed!
W (176777) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to send file
W (187547) httpd_txrx: httpd_sock_err: error in send : 11
W (192547) httpd_txrx: httpd_sock_err: error in send : 11
W (192547) httpd_uri: httpd_uri: uri handler execution failed
```

![Screenshot_20240129_013905](https://github.com/jomjol/AI-on-the-edge-device/assets/1671119/e39af15f-1051-462f-848f-9e04f5216f9b)
![Screenshot_20240129_013634](https://github.com/jomjol/AI-on-the-edge-device/assets/1671119/045d53b6-fe6e-4e34-ada9-36bb232da40a)

**Reply by arkoode** [reply]
Kann es sein das http die Requests nicht abarbeiten kann? Sprich, wenn zu viele auf einmal geschickt werden?

**Reply by caco3** [reply]
versuchs mal mit einer anderen SD-Karte und mit einem anderen Webbrowser

**Reply by kaiduerkop** [reply]
Habe 4 verschiedene ESPs von 2 Hersteller durch.
3 SD Karten von 3 verschiedenen Hersteller (2x 16GB und einmal eine 8GB)
Firefox, Edge, Opera - Überall gleiche Problem.

Mit curl wird die Seite zwar geladen, jedoch dauert es über eine Minute
Ich bin echt ratlos
```
> time curl http://192.168.105.115
<!DOCTYPE html>
<html lang="en" xml:lang="en">
<head>
    <link rel="icon" href="favicon.ico?v=74d4f20" type="image/x-icon">
[...]
</body>
</html>

real    1m16,768s
user    0m0,001s
sys     0m0,007s
```

**Reply by arkoode** [reply]
Wie hast du das Image geflasht?

**Reply by kaiduerkop** [reply]
Am Anfang über den AI on the Edge Webflasher (mittels VirtualBox USB Passthrough, da ich einen Linux Client habe).
Danach über das esptool
```
python -m esptool --chip esp32 write_flash 0x01000 bootloader.bin 0x08000 partitions.bin 0x10000 firmware.bin
```

**Reply by kaiduerkop** [reply]
Wie reagiert der ESP eigentlich auf ICMP Pakete?
Werden diese niedrig priorisiert?
Es gibt teilweise bis zu 5-30 Sekunden ICMP timeouts
Avg. ICMP ist bei 150-200ms. Teilweise 400ms.

Der ESP steht jedoch Luftlinie ~2m von einem Unifi AP entfernt welcher mittels PoE an einem Unifi Switch hängt.
Keine Ahnung ob das auch mit den Problemen zusammenhängt.

**Reply by mathiasmora** [reply]
@kaiduerkop: Hab genau das gleiche Problem. Die GUI lädt nicht vollständig und ich bekomme diverse Timeouts: 
```
I (43599) SNTP: =================================================
I (43599) SNTP: ==================== Start ======================
I (43599) SNTP: == Logs before time sync -> log_1970-01-01.txt ==
I (43599) SNTP: Time is synced with NTP Server pool.ntp.org: 2024-01-30 16:24:16
W (198219) httpd_txrx: httpd_sock_err: error in send : 11
E (198219) SERVER HELP: File sending failed!
W (203219) httpd_txrx: httpd_sock_err: error in send : 11
W (203219) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to send file
W (208219) httpd_txrx: httpd_sock_err: error in send : 11
W (213219) httpd_txrx: httpd_sock_err: error in send : 11
W (213219) httpd_uri: httpd_uri: uri handler execution failed
W (225219) httpd_txrx: httpd_sock_err: error in send : 11
E (225219) SERVER HELP: File sending failed!
W (230219) httpd_txrx: httpd_sock_err: error in send : 11
W (230219) httpd_txrx: httpd_resp_send_err: 500 Internal Server Error - Failed to send file
W (235219) httpd_txrx: httpd_sock_err: error in send : 11
W (240219) httpd_txrx: httpd_sock_err: error in send : 11
W (240219) httpd_uri: httpd_uri: uri handler execution failed
```

Habe auch schon unterschiedliche ESP32 Module verwendet und immer das gleiche Problem. Was mich gerade stutzig macht ist, dass ich auch ein Unifi System habe. Ggf. gibt es hier einen Zusammenhang bzw. Problem.

**Reply by bastiitsab** [reply]
hat hier jemand eine Lösung gefunden? Habe das gleiche Problem :(

**Reply by kaiduerkop** [reply]
Jo, habe mir ein neues Board besorgt mit einer WLAN Antenne, seit dem läuft es mehr oder weniger fehlerfrei.

**Reply by fsedarkalex** [reply]
Gleiches Problem, auch Unifi/Ubiquiti WLAN System... Ich sehe ein Muster...

**Reply by SybexX** [reply]
habt ihr WPA3 aktiviert?
https://ubiquiti-networks-forum.de/board/thread/4018-unifi-wlan-ist-eine-katastrophe/

**Reply by fsedarkalex** [reply]
ich nicht

**Reply by bastiitsab** [reply]
Ein Muster kann ich nicht bestätigen.
Es lag offenbar einfach an bescheidenen ESPs.
Ich hab mir 2 in anderen Shops bestellt und die laufen problemlos. Auch mit unifi.

**Reply by fsedarkalex** [reply]
We will see. Ich hab mir jetzt auch nochmal andere geordert.

**Reply by caco3** [reply]
Ich habe 2 ESP-CAMs am laufen, auch an Unifi Accesspoints, und habe keine solche Probleme!
Ich habe den ESPs externe Antennen spendiert, dafür müssen sie aus dem Keller in den 1. Stock hochsenden. RSSI ist mit 70 dBm gerade noch ok.

Ich würde das also nicht einfach auf die APs abschieben! Probleme machen eher die Kanalwahl und ausrichtung.

**Reply by SybexX** [reply]
naja wenn man den AP nicht richtig eingerichtet hat, kann es ja am AP oder doch am Benutzer liegen???? ist halt eine Auslegungssache^^
Und aus den Link weiter oben, den ich hier gepostet habe, geht hervor dass die Einstellung des Wlans nicht so einfach sein soll bzw. man viele Fehler dabei machen kann.

**Reply by caco3** [reply]
ja, aber ich wollte darauf hinweisen, dass es kaum direkt an Unifi liegt. Und ich verwende da weitestgehend die Standard config.

**Reply by SybexX** [reply]
meine Antwort und der Link weiter oben sollte eigentlich deine Aussage bekräftigen^^

**Reply by fsedarkalex** [reply]
Die zwei neuen gehen auch nicht. Jetzt hab ich nochmal 2 von berrybase bestellt. Wenn die auch nicht gehen geb ich auf. Das wird mir dann zu doof. (Da kann das Projekt aber vermutlich nichts dafür - die Hardware ist halt nicht gut konstruiert)

**Reply by Slider0007** [reply]
Hast du folgendes bereits probiert / beachtet. Anbei ein paar Hinweise: https://github.com/jomjol/AI-on-the-edge-device/discussions/2406#discussioncomment-8348127

**Reply by fsedarkalex** [reply]
Daumen bringt nur subjektiv minimal bessere Reaktionszeit aber keine Veränderung der Problematik
Netzteil ja
4 verschiedene SDs (2x Billig, SanDisk, Samsung) in 3 verschiedenen Größen
Verschiedene Kanäle
Handyhotspot nicht probiert, da seh ich keinen Sinn (muss ja später auch mit dem normalen WLAN laufen)

**Reply by Slider0007** [reply]
Dann hast du wohl Pech mit deinen Exemplaren und es liegt wirklich an der Hardware.

Externe Antenne ist noch eine weitere Möglichkeit.
Handy Hotspot hilft um AP Themen auszuschließen.

**Reply by fsedarkalex** [reply]
Du hast mich grad auf ne IDee gebracht... mit Erfolg!

Die Firmware kommt nicht damit klar, dass mein PC in einem anderen VLAN hängt (routing)...
Sobald die Kommunikation übers Gateway läuft geht nicht mehr... Bin ich aber im selben Netz läufts...

**Reply by fsedarkalex** [reply]
Hmm seltsam... irgendwie auch das nicht ganz... Absolut kurios... Es ist der Virenscanner! Mein Test mit dem anderen LAN war mit nem Gerät, das keinen hatte... Jetzt gings am Pc trotzdem nciht und ich hab weiter rumprobiert... Ich hab hier Kaspersky Internet Security drauf - Deaktiviert und zack läufts

**Reply by fsedarkalex** [reply]
Ich möchte  - nach einiger Zeit des Testlaufs - noch hinzufügen, dass sich das Problem auf den Setup-Assistenten beschränkt. Die Berichte vom Kaspersky sind allerdings sauber, also es ist kein offizieller "Block"... Wohl nur ne Inkompatibilität

**Reply by caco3** [reply]
> Wohl nur ne Inkompatibilität

Was ich äusserst interessant finde. Das Gerät ist ja einfach ein Webserver, so wie eine Fritzbox oder andere IoT-Geräte. Wieso sollte unser Gerät Probleme machen und andere nicht?

**Reply by fsedarkalex** [reply]
Virenscanner hängen sich in den Datenstrom und allein dadurch können sie Probleme verursachen. Ich mein der Webserver ist jetzt auch nicht der performanteste/beste weils halt ein minimalistischer auf einem ESP ist (Was völlig okay ist).

Das können verschiedenste dinge sein. Das Kaspersky nichts im Bericht meldet, hat er nicht blockiert, sondern es ist irgendeine Inkompatibilität. Aber schalte ich ihn ab, läufts. Also muss es daran liegen.

**Comment by Slider0007** [comment]
In der Regel ist das meistens zurückzuführen auf Hardware- oder WLAN Themen. Hierzu gibt es bereits viele Einträge mit teilweisen guten Tips.

- Hardware Antennendesign sehr schlecht in Verbindung mit Kamerafrequenz.
-> Einfacher Test: Bei laufenden Betrieb Daumen auf den Kamerastecker und die direkt angrenzenden Bauteile sollte das Gerät zügiger reagieren lassen.
-> Mögliche Optimierung: Hier wurde durch Abschirmung versucht die Frequenzeinflüsse etwas zu dämpfen. Das habe ich aber selbst nie probiert...

https://www.reddit.com/r/esp32/comments/r9g5jc/fixing_ymmv_the_poor_frame_rate_on_the_esp32cam/

- Ausreichend dimensioniertes Netzteil

- Marken SD Karte verwenden (formatiert mit Windows, MAC bereitet oft Probleme)

- WLAN Kanal: bevorzugt 1,6 oder 11
- Je nach AP Hersteller teilweise unterschiedliche Performance. Ggf. mit Handy-Hotspot oder anderem Gerät prüfen, um AP Einfluss auszuschließen 
- Externe Antenne verwenden
- VLAN vermeiden, wird aktuell nicht voll unterstützt 
- Virenscanner / Firewall auf Endgerät zum Test vorübergehend deaktivieren 
- ...

---

This is usually due to hardware or WLAN issues. There are already many entries on this, some of which have good tips.

- Hardware antenna design very poor in connection with camera frequency.
-> Simple test: When the device is in operation, putting your thumb on the camera connector and the directly adjacent components should make the device respond more quickly.
-> Possible optimization: Here, an attempt was made to dampen the frequency influences somewhat by shielding. But I've never tried that myself...
https://www.reddit.com/r/esp32/comments/r9g5jc/fixing_ymmv_the_poor_frame_rate_on_the_esp32cam/

- WLAN channel: Preferably use 1, 6 or 11

- Performance can vary depending on the AP manufacturer.  If necessary, check with a mobile hotspot or other device to exclude AP influence

- Use an external antenna

- Avoid VLAN, currently not fully supported

- Temporarily deactivate virus scanner / firewall on the end device for testing

- Sufficiently dimensioned power supply

- Use a branded SD card (formatted with Windows, MAC often causes problems)

**Reply by kaiduerkop** [reply]
> Hardware Antennendesign sehr schlecht in Verbindung mit Kamerafrequenz.
Einfacher Test: Bei laufenden Betrieb Daumen auf den Kamerastecker und die direkt angrenzenden Bauteile sollte das Gerät zügiger reagieren lassen.

Genau das war/ist es, eine Lösung dafür, um dieses eingebaut abzuschirmen weiß ich noch nicht, dazu muss ich mich nun noch mit google beschäften :)

**Reply by Slider0007** [reply]
Hier wurde durch Abschirmung versucht die Frequenzeinflüsse etwas zu dämpfen. Das habe ich aber selbst nie probiert...

https://www.reddit.com/r/esp32/comments/r9g5jc/fixing_ymmv_the_poor_frame_rate_on_the_esp32cam/

**Reply by kaiduerkop** [reply]
Ich habe mir von einem Bekannten noch eine andere ESP32 Cam besorgt.
Funktioniet tadellos... Und dabei wurde, die ich hatte, auf Amazon angepriesen als "funktioniert mit AI on the edge"

**Reply by caco3** [reply]
"funktioniert mit AI on the edge" ist ein genereller Term, das ist viel mehr als nur unser Gerät.
Zudem, wenn der Händler aus China kommt, kannst Du seinen Angaben eh nicht trauen.

**Reply by kaiduerkop** [reply]
Ich hatte erwartet das AZ Delivery gute Boards herstellt, die hatten/haben sogar mit die meisten positiv Bewertungen, naja. so kann man sich täuschen :)

**Reply by caco3** [reply]
AZ Delivery stellt keine Boards her, die importieren sie auch einfach von China.
